### PR TITLE
CB-4904 test report improvements

### PR DESCRIPTION
**Pull request by afarsang**
test report improvements
Closes 4904

### CODE REVIEW

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/log/Log.java [hunk 1]
```diff
@@ -4,28 +4,30 @@
...
+    public static final String TEST_CONTEXT_REPORTER = "testContextReporter";
+
+    public static final String ALTERNATE_LOG = "alternateLog";
 
-public class Log {
     private Log() {
     }
 
-    public static void logJSON(Logger logger, String message, Object jsonObject) throws IOException {
+    public static void logWhenJson(Logger logger, String message, Object jsonObject) throws IOException {
```
Comment (topolyai5) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/log/Log.java]: I think whenJson is enough like the others methods (when, then..)

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/log/Log.java [hunk 2]
```diff
@@ -40,7 +42,67 @@ public static void log(Logger logger, String message, Object... args) {
...
+        return reporter;
+    }
+
+    private static void publishReport(TestContextReporter testContextReporter) {
+        ITestResult res = Reporter.getCurrentTestResult();
+        if (res == null) {
+            return;
+        }
+        res.setAttribute(ALTERNATE_LOG, testContextReporter.print());
+    }
```
Comment (topolyai5) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/log/Log.java]: You can use negation and add to the if block the operation

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/log/TestContextReporter.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    public void addStep(String step, String description) {
+        steps.add(new Step(step, description));
+    }
+
+    public void addStep(String step, String description, String json) {
+        steps.add(new Step(step, description, json));
+    }
+
+    public String print() {
+        if (steps.size() == 0) {
```
Comment (topolyai5) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/log/TestContextReporter.java]: Please use .isEmpty()

Comment (afarsang) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/log/TestContextReporter.java]: Ok

**Pull request merged**